share/lib/python/neuron/coreneuron.py

```diff
@@ -1,92 +1,214 @@
+
+    @property
+    def num_gpus(self):
+        """Get/set the number of GPU to use.
+        0 means to use all that are avilable.
+        """
+        return self._num_gpus
+
+    @num_gpus.setter
+    def num_gpus(self, value):
+        self._num_gpus = int(value)
+
+    @property
+    def file_mode(self):
+        """Run via file transfer mode instead of in-memory transfer."""
+        return self._file_mode
+
+    @file_mode.setter
+    def file_mode(self, value):
+        self._file_mode = bool(value)
+
+    @property
+    def cell_permute(self):
+        """Get/set the cell permutation order. Possible values are:
+        - 0: no permutation (requires gpu = False)
+        - 1: optimize node adjacency
+        - 2: optimise parent node adjacency (requires gpu = True)
+        """
+        if self._cell_permute is None:
+            return self._default_cell_permute()
+        return self._cell_permute
+
+    @cell_permute.setter
+    def cell_permute(self, value):
+        value = int(value)
+        assert value in self._valid_cell_permute()
+        self._cell_permute = value
+
+    @property
+    def warp_balance(self):
+        """Number of warps to balance. A value of 0 indicates no balancing
+        should be done. This has no effect without cell_permute = 2 (and
+        therefore gpu = True)."""
+        return self._warp_balance
+
+    @warp_balance.setter
+    def warp_balance(self, value):
+        value = int(value)
+        assert value >= 0
+        self._warp_balance = value
+
+    @property
+    def verbose(self):
+        """Configure the verbosity of CoreNEURON. Possible values are:
+        - 0: quiet/no output
+        - 1: error
+        - 2: info (default)
+        - 3: debug
+        """
+        return self._verbose
+
+    @verbose.setter
+    def verbose(self, value):
+        value = int(value)
+        assert value >= 0 and value <= 3
+        self._verbose = value
+
+    @property
+    def prcellstate(self):
+        """Output prcellstate information for the given gid at t=0 and at tstop.
+        A value of -1 indicates no prcellstate output will be generated.
+        """
+        return self._prcellstate
+
+    @prcellstate.setter
+    def prcellstate(self, value):
+        value = int(value)
+        assert value >= -1
+        self._prcellstate = value
+
+    def nrncore_arg(self, tstop):
+        """
+        Return str that can be used for pc.nrncore_run(str)
+        based on user properties and current NEURON settings.
+        :param tstop: Simulation time (ms)
+
+        """
+        from neuron import h
+
+        arg = ""
+        tstop = float(tstop)
+        if not self.enable:
+            return arg
+
+        # note that this name is also used in C++ file nrncore_write.cpp
+        CORENRN_DATA_DIR = "corenrn_data"
+
+        # args derived from user properties
+        if self._gpu:
+            arg += " --gpu"
+            if setf._num_gpus:
```

Comment:
```suggestion
            if self._num_gpus:
```
and it needs to be rebased now